File: C# Labs 2 semester/6Lab_Programming_Polimorphism-master/Classes.cs
using System;

namespace Lab6Polymorphism
{
    abstract public class Lines
    {
        abstract public int Length();
        abstract public void Decrement();
    }
    public class Symbols : Lines
    {
        private string str;
        public Symbols(string _str)
        {
            str = _str;
        }
        public override int Length()
        {
            return str.Length;
        }
        public override void Decrement()
        {
            string NewStr = "";
            int count = 1;
            while (count < str.Length)
            {
                NewStr += str[count];
                count += 2;
            }
            str = NewStr;
        }
        public string GetData()
        {
            return str;
        }
    }

    public class Numbers : Lines
    {
        private string str;
        public Numbers(string _str)
        {
            str = _str;
        }
        public override int Length()
        {
            return str.Length;
        }
        public override void Decrement()
        {
            string NewStr = "";
            int count = 0;
            while (count < str.Length)
            {
                NewStr += str[count];
                count += 2;
            }
            str = NewStr;
        }
        public string GetData()
        {
            return str;
        }
    }

}

File: C# Labs 2 semester/6Lab_Programming_Polimorphism-master/Classes.cpp
#include "Header.h"
Symbols::Symbols(string _str)
{
	str = _str;
}

int Symbols::Length()
{
	return str.length();
}

void Symbols::Decrement()
{
    string NewStr = "";
    int count = 1;
    while (str[count] != '\0')
    {
        NewStr += str[count];
        ++count;
        if (str[count] == '\0')
        {
            break;
        }
        ++count;
    }
    str = NewStr;
}

string Symbols::GetData()
{
	return str;
}

Numbers::Numbers(string _str)
{
	str = _str;
}

int Numbers::Length()
{
	return str.length();
}

void Numbers::Decrement()
{
    string NewStr = "";
    int count = 0;
    while (str[count] != '\0')
    {
        NewStr += str[count];
        ++count;
        if (str[count] == '\0')
        {
            break;
        }
        ++count;
    }
    str = NewStr;
}

string Numbers::GetData()
{
	return str;
}